packages/core/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -527,14 +528,26 @@ export async function pollTransformationStatusUntilPlanReady(jobId: string) {
         // Since we don't yet have a good way of knowing what the error was,
         // we try to fetch any build failure artifacts that may exist so that we can optionally
         // show them to the user if they exist.
-        const tmpDir = path.join(os.tmpdir(), 'q-transformation-build-logs')
-        await downloadAndExtractResultArchive(jobId, undefined, tmpDir, 'Logs')
-        const pathToLog = path.join(tmpDir, 'buildCommandOutput.log')
-        transformByQState.setPreBuildLogFilePath(pathToLog)
+        let pathToLog = ''
+        try {
+            const tempToolkitFolder = await makeTemporaryToolkitFolder()
```

Comment:
Here's a simpler solution to solve this problem: use `makeTemporaryToolkitFolder()` instead of `os.tmpdir()` so that a new folder with a random name is used to download the logs, which removes the requirement to find and delete the old logs.

Reply:
> which removes the requirement to find and delete the old logs.

is that causing problems? ideally we should not leave things in /tmp, even though in theory the OS cleans it up (in practicae not really)

Reply:
We were previously using `os.tmpdir()` which appears to stay constant, so we had to delete the old logs. When we downloaded the new logs, for some reason that ZIP was not overwriting the old ZIP, so the old logs were showing up. We need to be able to store the logs indefinitely after a transformation job so the user can click on them anytime until they run a new transformation.

Reply:
Anytime `makeTemporaryToolkitFolder()` is used there are going to be things left there undeleted, unless I store the random names of the all of the temporary folders created for an arbitrary number of transformations, and clean them up myself in place of the OS.

Reply:
> We need to be able to store the logs indefinitely after a transformation job so the user can click on them anytime until they run a new transformation.

I see. Usually I'd suggest a workspace-local hidden folder for this, that is often what users expect. But then the user will need to add that to .gitignore.